lib/tarfiles.py

```diff
@@ -73,10 +73,9 @@ def check_we_can_write() -> None:
 
 def tarchmod(tfn: str) -> str:
     """copy a tarfile to a compressed tarfile changing modes of contents to 755"""
-    os.environ["GZIP"] = "-n"
-    ofn = os.path.basename(f"{tfn}.o.gz")
+    ofn = os.path.basename(f"{tfn}.o.bz2")
     check_we_can_write()
-    with tarfile_mod.open(tfn, "r|*") as fin, tarfile_mod.open(ofn, "w|gz") as fout:
+    with tarfile_mod.open(tfn, "r|*") as fin, tarfile_mod.open(ofn, "w|bz2") as fout:
```

Comment:
The idea here is that bz2 gives us hash reproducibility independent of time, and gz does not?

Reply:
Exactly.  I *thought* it would get fixed with the same GZIP=-n in the environment that works for actual /usr/bin/tar, but no such luck.

Reply:
Yeah - I believe this is what jobsub_client used to do, so I'm surprised it doesn't work here.  What's the effect on currently-uploaded tarballs?  The hashes will all be different than before, so basically every single new submission using dropbox:// or tardir:// will upload a new tarball to RCDS, right?